src/cpp/rtps/builtin/discovery/database/DiscoveryDataBase.cpp

```diff
@@ -136,22 +136,60 @@ void DiscoveryDataBase::clear_edp_subscriptions_to_send()
 
 bool DiscoveryDataBase::process_data_queue()
```

Comment:
Call this method in `PDPServer2::server_update_routine()`. Be careful cause this returns whether there are dirty topics, but `PDPServer2` only cares about that after processing them, so you should probably ignore the return of `DiscoveryDataBase::process_data_queue()` in `PDPServer2::server_update_routine()`